File: 20241022/minseo/동영상재생기.java
class Solution {
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        String answer = "";
        
        int max = getTime(video_len);
        int start = getTime(op_start);
        int end = getTime(op_end);
        int current = getTime(pos);
        
        // 재생 중인 경우
        if(start <= current && current <= end){
            current = end;
        }
        
        for(String command : commands){
            if(command.equals("next")){
                current += 10;
            }
            else{
                current -= 10;
            }
            
            if(current < 0){
                current = 0;
            }
            
            else if(current > max){
                current = max;
            }
            
            if(start <= current && current <= end){
                current = end;
            }
        }
        // gpt 참고함 ㅎ..
        return String.format("%02d:%02d", current / 60 , current % 60);
    }
    
    int getTime(String time){
        // 분으로 바꿔주기
        String[] arr = time.split(":");        
        return Integer.parseInt(arr[1]) + Integer.parseInt(arr[0]) * 60;
    }
}

File: 20241021/haeun/동영상재생기.js
function solution(video_len, pos, op_start, op_end, commands) {
    const videoLen = changeSeconds(video_len)
    let posTime = changeSeconds(pos)
    const opStart = changeSeconds(op_start)
    const opEnd = changeSeconds(op_end)

    if (posTime >= opStart && posTime <= opEnd) posTime = opEnd;

    for (const command of commands) {
        if (command === "next") posTime = goNext(posTime);
        if (command === "prev") posTime = goPrev(posTime);
        if (posTime >= opStart && posTime <= opEnd) posTime = opEnd;
    }

    return changeForm(posTime);

    function changeSeconds(str) {
        const arr = str.split(":").map(Number)
        return arr[0] * 60 + arr[1];
    }

    function goNext(time) {
        if (time > videoLen - 10) return videoLen;
        return time + 10;
    }
    function goPrev(time) {
        if (time < 10) return 0;
        return time - 10;
    }


    function changeForm(time) {
        let minute = parseInt(time / 60).toString();
        let second = (time % 60).toString();
        if (minute < 10) minute = "0" + minute;
        if (second < 10) second = "0" + second;

        return `${minute}:${second}`
    }
}